src/main/java/duke/controller/TaskManager.java

```diff
@@ -0,0 +1,127 @@
+package duke.controller;
+
+import duke.exception.*;
+import duke.task.*;
+import duke.ui.*;
+
+public class TaskManager {
+    public static void processTask(String input) throws InvalidInputException {
+        String[] inputData = input.split(" ");
+        String[] taskDetails;
+        switch (inputData[0]) {
+            case "done":
+            case "delete":
+                try {
+                    int taskIndex = Integer.parseInt(input.split(" ")[1]);
+                    taskIndex--;
+                    Task.checkTaskIndex(taskIndex);
+                    if (inputData[0].equals("done")) {
```

Comment:
Try to avoid deep nesting. Refer to code quality

Reply:
Done! Thank you!